### PR TITLE
Filtered download

### PR DESCRIPTION
**Pull request by dianashk**
Connected to pelias/pelias#506

Adding download script that will support downloading the entire bundle or just selected files.
The existing `imports.openaddress.files` config parameter will be used to filter the download process as well as the import process.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +44,7 @@ other region like province), or country that they belong to. Pelias has the
 ability to compute these values from [Who's on First](http://whosonfirst.mapzen.com/) data.
 For more info on how admin lookup works, see the documentation for
 [pelias/wof-admin-lookup](https://github.com/pelias/wof-admin-lookup). By default,
-adminLookup is enabled.  To disable, set `imports.adminLookup.enabled` to `false` in Pelias config.
+adminLookup is enabled.  To disable, set `imports.openaddresses.adminLookup` to `false` in Pelias config.
```
Comment (orangejulius) [inline README.md]: If I remember correctly, we moved this flag, it's no longer per-importer, so the readme is correct.

File: utils/download_filtered.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    async.each(files, downloadSource.bind(null, targetDir), callback);
+
+  });
+
+}
+
+function downloadSource(targetDir, file, callback) {
+  logger.info(`Downloading ${file}`);
+
+  // first convert the file path to the source name, like 'us/ny/city_of_new_york.csv' -> 'us-ny-city_of_new_york.zip'
```
Comment (orangejulius) [inline utils/download_filtered.js]: the code doesn't convert slashes, but it looks like that's actually the correct format for the URL, so this comment should be updated, right?

File: utils/download_filtered.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const source = _.replace(file, '.csv', '.zip');
+  const sourceUrl = `https://results.openaddresses.io/latest/run/${source}`;
+  const tmpZipFile = tmp.tmpNameSync({postfix: '.zip'});
```
Comment (orangejulius) [inline utils/download_filtered.js]: I like that a tmp file is used. Do you think it's still worth it to clean out those files after they're unzipped? OA data can get pretty big...

File: README.md [hunk 4]
```diff
@@ -64,19 +71,42 @@ hash. A sample configuration file might look like:
...
+      "datapath": "/mnt/data/whosonfirst/",
+      "importPostalcodes": false,
+      "importVenues": false
+    },
     "openaddresses": {
-        "deduplicate": false,
-        "datapath": "/tmp/oa-data",
-        "files": ["us/ny/city_of_new_york.csv"]
-      }
+      "adminLookup": true,
```
Comment (orangejulius) [inline README.md]: This can also go away

File: README.md [hunk 5]
```diff
@@ -64,19 +71,42 @@ hash. A sample configuration file might look like:
...
-  * `files`: An array of the names of the files to import. If specified, *only* these files will be imported, rather
-    than *all* `.csv` files in the given directory. **If the array is empty, all files will be imported.**
+
+This importer is configured using the [`pelias-config`](https://github.com/pelias/config) module.
+The following configuration options are supported by this importer.
+
+| key | required | default | description |
+| --- | --- | --- | --- |
+| `deduplicate` | no | `false` | Boolean flag to enable deduplication (see above). |
+| `adminLookup` | no | `true` | Boolean flag to enable admin-lookup. |
```
Comment (orangejulius) [inline README.md]: this one too

**Comment by tigerlily-he** [comment]
The `README.txt` (listing the date the data was packaged, website source, and license) from openaddress.io gets downloaded along with the data. It's not a problem but just unnecessary?

File: README.md [hunk 6]
```diff
@@ -64,19 +71,42 @@ hash. A sample configuration file might look like:
...
+| `datapath` | yes | | The absolute path of the directory containing OpenAddresses files. Must be specified if no directory is given as a command-line argument. |
+| `files` | no | | An array of the names of the files to download/import. If specified, *only* these files will be downloaded and imported, rather than *all* `.csv` files in the given directory. **If the array is empty, all files will be downloaded and imported.** |
```
Comment (tigerlily-he) [inline README.md]: Someone new to OpenAddresses might not know how the files or directories are named. Add a link to http://results.openaddresses.io/?runs=all#runs here or under "Data Download"

File: package.json [hunk 7]
```diff
@@ -37,6 +40,7 @@
     "temp": "^0.8.3"
   },
   "scripts": {
+    "download": "node utils/download_data.js",
     "import": "node import.js",
```
Comment (tigerlily-he) [inline package.json]: This wasn't a change in this PR but I noticed an inconsistency: The README recommends  `node import.js`, while all the other importers call `npm start` to import. 

The only difference between `npm start` and `npm import` here is `--max_old_space_size=4096`in `npm start`

Comment (dianashk) [inline package.json]: good catch. I've updated here even though it's not part of this change.

**Comment by dianashk** [comment]
### Update:
Thanks all for the thoughtful feedback. I believe I've addressed all of it with the latest commits. Please have another look.

File: utils/download_all.js [hunk 8]
```diff
@@ -46,7 +46,9 @@ function downloadBundle(targetDir, sourceUrl, callback) {
       (callback) => {
         logger.debug(`unzipping ${tmpZipFile} to ${targetDir}`);
         child_process.exec(`unzip -o -qq -d ${targetDir} ${tmpZipFile}`, callback);
-      }
+      },
+      // delete the temp downloaded zip file
+      fs.remove.bind(null, tmpZipFile)
```
Comment (orangejulius) [inline utils/download_all.js]: this is pretty fancy and I love how async.series makes the flow clear here :)
